### DISCUSSION TITLE
dropdown list using datapane

### DISCUSSION THREAD

**Discussion by ashishlal** [datapane/datapane#124] [Help]
I want to publish a plot with dropdown list using datapane. Is it possible? Can someone show me with an example how it can be done?

**Comment by isergii** [comment]
@ashishlal , you can use Bokeh for this purpose. Check and try examples from here:
https://www.tutorialspoint.com/bokeh/bokeh_adding_widgets.htm

So, generate plot with interactive elements (dropdown, radio buttons, etc.) and then insert it in datapane report.

**Reply by ashishlal** [reply]
Thanks for the response. This is awesome. I was looking for something similar. I am trying to use plotly graph objects to create a drop down list. I want to display the usage statistics of my app (weekly, monthly, yearly etc).
Is there a way to connect a date picker range to a datapane webpage? For example, I want to select start date, end date, then use dropdown list to select "weekly" or "monthly" and then display the statistics accordingly.

**Reply by isergii** [reply]
Datapane is doing a great job by generating html reports, but it does not provide the same flexibility as you would have with for example Plotly Dash. By flexibility I mean higher interactivity, data flow between components, callbacks support etc. - but it requires running server and solving other issues if you want to share dashboard with several people.

Datapane's html reports are already very good in the sense of interactivity due to Bokeh, Plotly support. The only thing that you need to take care about is to combine all your visualization logic into single "fig" for example, and then use dp.Plot(fig).

For example if you take example from here:
https://plotly.com/python/sliders/
and put resulting fig into dp.Plot(), it will work. Then you can add more improvements (sliders, dropdowns, everything you find  in Plotly) to your visualization and so on.

I personally found Bokeh more promising in this scope:
https://docs.bokeh.org/en/latest/docs/user_guide/interaction/widgets.html
You can construct really sophisticated layouts with interactive components in Bokeh and then just do the same - dp.Plot().

**Reply by mands** [reply]
Thanks @isergii for that detailed description! As mentioned, yes Datapane focussed more on generating standalone HTML reports that don't require a running server. Most interactively currently is done via the plotting library, e.g. `plotly`, `altair`, `bokeh`, etc. - a lot of these have very advanced functionality for doing more interactive analysis within the plot itself.

An alternate approach that could work for you is to use `dp.Select` to allow users to switch between various precomputed plots.

Finally, we are also looking at some ways we can provide more interactively directly from the browser, but that is a few months away at the moment. Datapane Teams does also provide support for running creating dynamic reports based on user input if you are in an organisation - we are working on some ways to support this for local users too.

HTH!
